client/influxdb.go

```diff
@@ -64,23 +65,36 @@ type Query struct {
 	NodeID int
 }
 
+func splitPath(v string) (string, string) {
+	parts := strings.Split(v, "/")
+
+	first := parts[0]
+	last := strings.Join(parts[1:], "/")
+
+	return first, last
+}
+
 // ParseConnectionString will parse a string to create a valid connection URL
 func ParseConnectionString(path string, ssl bool) (url.URL, error) {
 	var host string
 	var port int
+	var pth string = ""
 
 	h, p, err := net.SplitHostPort(path)
 	if err != nil {
 		if path == "" {
 			host = DefaultHost
 		} else {
-			host = path
+			host, pth = splitPath(path)
```

Comment:
Can probably use `url.Parse` to get this information without manually splitting the string. Might be safer / more readable.

Reply:
it won't always be a valid url because it could be `host/path` or `port/path` in which case parse would fail with an error.

Reply:
The client allows connect strings like `localhost:8086` that `url.Parse` can't parse but also won't error. Relevant blurb from Go docs:

> Trying to parse a hostname and path without a scheme is invalid but may not necessarily return an error, due to parsing ambiguities.

Reply:
@dgnorton I am curious why we allow host and port together when it is not explicitly stated in the host flag usage? This could be a lot cleaner if we had `host` `port` `urlpath` flags and a `connection string` flag for the combination of the 3

Reply:
I think we allow them together for the `connect` command (allowing you to connect to another influxdb from within the CLI)

Reply:
then connection string and the connect command could share logic while allowing the other flags to be independent, there is a bit of tangle logic that could be cleaned up by treating it this way I think. Right now I could supply a port from the `port` flag or maybe I give it in the `host` where as if we had a `connection string` flag we could say, if this is present get port from it and ignore other flags (maybe even give invalid options error if port and connection string are given together.

Reply:
maybe instead of `connection string` we say `url` and then we can use `url.Parse` on it

Reply:
A single command line connection flag that is flexible would be nice:
`http://localhost:8086` (default)
`localhost`
`:9086`
`http://somewhere.com`
`http://somewhere.com:9096`
`http://somewhere.com/influxdb`
`https://somewhere.com:9086/influxdb`
etc.

Need to ensure existing flags continue to work as before. Some users have scripts that depend on existing behavior.